scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/ProgressFileReader.java

```diff
@@ -136,19 +141,31 @@ public void unregister(Listener listener) {
     public void stop() {
         if (watchService != null) {
             try {
-                watchService.close();
-                watchServiceThread.join();
+                closeWatchService();
             } catch (InterruptedException e) {
                 Thread.currentThread().interrupt();
                 logger.info(e);
-            } catch (IOException e) {
+            } catch (Exception e) {
                 logger.error(e);
             } finally {
                 removeProgressFileDir();
             }
         }
     }
 
+    private void closeWatchService() throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)